webgl.js

```diff
@@ -3482,6 +3482,36 @@ function checkFormat (format) {
     format === gl.RGBA)
 }
 
+var extractImageData = function (pixels) {
+  if (typeof pixels === 'object' && typeof pixels.width !== 'undefined' && typeof pixels.height !== 'undefined') {
+    if (typeof pixels.data !== 'undefined') {
+      return pixels
+    }
+
+    var context = null
+
+    if (typeof pixels.getContext === 'function') {
+      context = canvas.getContext('2d')
+    } else if (typeof pixels.src !== 'undefined' && typeof document === 'object' && typeof document.createElement === 'function') {
+      var canvas = document.createElement('canvas')
+
+      if (typeof canvas === 'object' && typeof canvas.getContext === 'function') {
+        context = canvas.getContext('2d')
+
+        if (context !== null) {
+          context.drawImage(pixels, 0, 0)
+        }
+      }
+    }
+  }
+
+  if (context !== null) {
```

Comment:
@gre Whoops, quite right. Originally I had the `context` variable declaration outside the outer `if` statement but moved it just before submitting the pull request.

Force pushing incoming...